packages/alchemy/src/client/rpcClient.ts

```diff
@@ -17,16 +43,25 @@ export const createAlchemyPublicRpcClient = ({
       ? `${chain.rpcUrls.alchemy.http[0]}/${connectionConfig.apiKey ?? ""}`
       : connectionConfig.rpcUrl;
 
+  const dynamicFetchOptions = fetchOptions ?? createDynamicFetchOptions();
+
+  dynamicFetchOptions.headers = {
+    ...dynamicFetchOptions.headers,
+    "Alchemy-AA-Sdk-Version": VERSION,
+    ...(connectionConfig.jwt != null
+      ? { Authorization: `Bearer ${connectionConfig.jwt}` }
+      : undefined),
+  };
+
   return createBundlerClient({
     chain: chain,
-    transport: http(rpcUrl, {
-      ...(connectionConfig.jwt != null && {
-        fetchOptions: {
-          headers: {
-            Authorization: `Bearer ${connectionConfig.jwt}`,
-          },
-        },
-      }),
-    }),
-  });
+    transport: http(rpcUrl, { fetchOptions: dynamicFetchOptions }),
```

Comment:
viem's transport takes the fetch options and passes them down to the various functions that consume them. but it does so by doing things like 
```
const httpTransport = args => {
  const { fetchOptions } = args; 
  return () => fetchOptions.foo;
}
```